src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,71 @@
 package core.basesyntax;
 
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.List;
+
 public class WorkWithFile {
+    private final StringBuilder array = new StringBuilder();
+    private boolean dataWritten = false;
+
     public void getStatistic(String fromFileName, String toFileName) {
+        readFromFile(fromFileName);
+        writeToFile(toFileName);
+    }
+
+    private void readFromFile(String fileName) {
+        File file = new File(fileName);
+        try {
+            List<String> strings = Files.readAllLines(file.toPath());
+            String text = String.join(System.lineSeparator(), strings);
+            String validText = text.replace(System.lineSeparator(), ",");
+            String[] value = validText.split(",");
+            createReport(value);
```

Comment:
read from file shouldn't deal with report creation. Make it return value, and call create report on that value in getStatistic public method